Lesson17/Ayse/contactBook.js

```diff
@@ -0,0 +1,331 @@
+
+You'll practice:
+1. Arrays and objects
+2. Loops (for, for-of, findIndex, etc.)
+3. Conditionals (if-else)
+4. Basic CRUD (Create, Read, Update, Delete) functionality
+
+Below is a step-by-step guide, with sample code and
+comments explaining what each section does. Run this file
+in Node.js or in your browser's console to see the output.
+*/
+
+/*
+-----------------------------------------------------------
+  STEP 1: Setup and Initial Data
+-----------------------------------------------------------
+1. Create an array named 'contacts' with a few initial
+   sample contacts.
+2. Each contact is an object with 'name', 'phone', and
+   'email' properties.
+*/
+console.log('STEP-1: Setup and Initial Data');
+
+let contacts = [
+  {
+    name: 'Alice',
+    phone: '123-456-7890',
+    email: 'alice@gmail.com'
+  },
+  {
+    name: 'Anton',
+    phone: '123-456-7899',
+    email: 'anton@gmail.com'
+  },
+  {
+    name: 'Bob',
+    phone: '222-222-2222',
+    email: 'bob@gmail.com'
+  }
+];
+console.log(contacts);
+
+/*
+-----------------------------------------------------------
+  STEP 2: Display All Contacts
+-----------------------------------------------------------
+Function: displayAllContacts()
+- Loops over the 'contacts' array.
+- Logs a descriptive string for each contact.
+
+Example output:
+  Name: Alice, Phone: 123-456-7890, Email: alice@example.com
+*/
+console.log('STEP 2: Display All Contacts');
+console.log('Method-1');
+
+function displayAllContacts(contactList){
+
+  for(let contact of contactList){
+    console.log(`Name: ${contact.name}, Phone: ${contact.phone}, Email: ${contact.email}`);
+  }
+}
+displayAllContacts(contacts);
+
+console.log('Method-2');
+
+function displayAllContactsUppercase(contactList, uppercaseMode){
+  for (const contact of contactList){
+    if(uppercaseMode){
+      console.log(`NAME: ${contact.name}, PHONE: ${contact.phone}, EMAIL: ${contact.email}`,);
+      continue;
+    }else{
+      console.log(`name: ${contact.name}, phone: ${contact.phone}, email: ${contact.email}`,);
+    }
+  }
+}
+displayAllContactsUppercase(contacts, true);
+displayAllContactsUppercase(contacts, false);
+
+/*
+-----------------------------------------------------------
+  STEP 3: Add a New Contact
+-----------------------------------------------------------
+Function: addContact(name, phone, email)
+- Creates a new contact object and pushes it into 'contacts'.
+- Checks if a contact with the same name already
+  exists before adding. If found, logs a warning and returns.
+- Logs "Contact added successfully." if everything is good.
+*/
+console.log('STEP 3: Add a New Contact');
+
+
+let contacts3 = [
+  { name: 'Alice', phone: '123-456-7890', email: 'alice@gmail.com' },
+  { name: 'Anton', phone: '123-456-7899', email: 'anton@gmail.com' },
+  { name: 'Bob', phone: '222-222-2222', email: 'bob@gmail.com' }
+];
+
+function addContact(name, phone, email, contactList) {
+  for (let i = 0; i < contactList.length; i++) {
```

Comment:
Many functions in this application need to find a contact in the list. For that, we created a findContact function in the lesson that checks if the contact is in the list or not. Use it here instead of the loop.